Travis CI issue on Mac OS

```
[  6%] Building CXX precompiled header symengine/cotire/symengine_CXX_prefix.hxx.pch

clang: error: invalid version number in '-mmacosx-version-min=10.13 -std=gnu++11'

CMake Error at /private/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/Rtmp7FAN6d/Rbuildd3cb2c0d5e0b/symengine/src/upstream/cmake/cotire.cmake:1826 (message):

  cotire: error 1 precompiling

  /private/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/Rtmp7FAN6d/Rbuildd3cb2c0d5e0b/symengine/src/upstream/symengine/cotire/symengine_CXX_prefix.hxx.

Call Stack (most recent call first):

  /private/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/Rtmp7FAN6d/Rbuildd3cb2c0d5e0b/symengine/src/upstream/cmake/cotire.cmake:3421 (cotire_precompile_prefix_header)

make[3]: *** [symengine/cotire/symengine_CXX_prefix.hxx.pch] Error 1

make[2]: *** [symengine/CMakeFiles/symengine.dir/all] Error 2

make[1]: *** [all] Error 2
```

Thanks for reporting it. Apparently in R-4.0, `R CMD config CXX` returns `clang++ -mmacosx-version-min=10.13 -std=gnu++11`, we set it as `CXX` environment variable and run cmake for the C++ library, which leads to this error.

I also tested the following values:
- Setting `CXX="clang++  -mmacosx-version-min=10.13"` runs fine.
- Setting `CXX="clang++ -std=gnu++11"` runs fine.
- Setting `CXX="clang++ -std=gnu++11 -mmacosx-version-min=10.13"` leads to
   ```
   [  3%] Building CXX object symengine/CMakeFiles/symengine.dir/add.cpp.o
   error: stack checking was enabled in PCH file but is currently disabled
   error: PCH file was compiled for the target 'x86_64-apple-macosx10.15.0' but the current translation
         unit is being compiled for target 'x86_64-apple-macosx10.13.0'
   2 errors generated.
   make[2]: *** [symengine/CMakeFiles/symengine.dir/add.cpp.o] Error 1
   make[1]: *** [symengine/CMakeFiles/symengine.dir/all] Error 2
   make: *** [all] Error 2
   ```

I think the current workaround would be just to remove the '-mmacosx-version-min' flag.

Thank you very much, Marlin. I check what you suggest and I go back to you.


Le mer. 6 mai 2020 à 13:14, Marlin NȺ <notifications@github.com> a écrit :

> Thanks for reporting it. Apparently in R-4.0, R CMD config CXX returns clang++
> -mmacosx-version-min=10.13 -std=gnu++11, we set it as CXX environment
> variable and run cmake for the C++ library, which leads to this error.
>
> I also tested the following values:
>
>    - Setting CXX="clang++ -mmacosx-version-min=10.13" runs fine.
>    - Setting CXX="clang++ -std=gnu++11" runs fine.
>    - Setting CXX="clang++ -std=gnu++11 -mmacosx-version-min=10.13" leads
>    to
>
>    [  3%] Building CXX object symengine/CMakeFiles/symengine.dir/add.cpp.o
>    error: stack checking was enabled in PCH file but is currently disabled
>    error: PCH file was compiled for the target 'x86_64-apple-macosx10.15.0' but the current translation
>          unit is being compiled for target 'x86_64-apple-macosx10.13.0'
>    2 errors generated.
>    make[2]: *** [symengine/CMakeFiles/symengine.dir/add.cpp.o] Error 1
>    make[1]: *** [symengine/CMakeFiles/symengine.dir/all] Error 2
>    make: *** [all] Error 2
>
>
> I think the current workaround would be just to remove the
> '-mmacosx-version-min' flag.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/symengine/symengine.R/issues/88#issuecomment-624587519>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APJI4DP7EKHNJMV4HITUZWTRQFA7TANCNFSM4MX55WWQ>
> .
>

Dear Marlin

It has worked fine. Thanks

Christian


Le mer. 6 mai 2020 à 13:14, Marlin NȺ <notifications@github.com> a écrit :

> Thanks for reporting it. Apparently in R-4.0, R CMD config CXX returns clang++
> -mmacosx-version-min=10.13 -std=gnu++11, we set it as CXX environment
> variable and run cmake for the C++ library, which leads to this error.
>
> I also tested the following values:
>
>    - Setting CXX="clang++ -mmacosx-version-min=10.13" runs fine.
>    - Setting CXX="clang++ -std=gnu++11" runs fine.
>    - Setting CXX="clang++ -std=gnu++11 -mmacosx-version-min=10.13" leads
>    to
>
>    [  3%] Building CXX object symengine/CMakeFiles/symengine.dir/add.cpp.o
>    error: stack checking was enabled in PCH file but is currently disabled
>    error: PCH file was compiled for the target 'x86_64-apple-macosx10.15.0' but the current translation
>          unit is being compiled for target 'x86_64-apple-macosx10.13.0'
>    2 errors generated.
>    make[2]: *** [symengine/CMakeFiles/symengine.dir/add.cpp.o] Error 1
>    make[1]: *** [symengine/CMakeFiles/symengine.dir/all] Error 2
>    make: *** [all] Error 2
>
>
> I think the current workaround would be just to remove the
> '-mmacosx-version-min' flag.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/symengine/symengine.R/issues/88#issuecomment-624587519>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APJI4DP7EKHNJMV4HITUZWTRQFA7TANCNFSM4MX55WWQ>
> .
>

--- a/.travis.yml
+++ b/.travis.yml
@@ -34,7 +34,7 @@ apt_packages:
 #  - libmpc
 
 before_install:
-  - if [ $TRAVIS_OS_NAME = osx ]; then brew install cmake gmp mpfr libmpc || true; fi
+  - if [ $TRAVIS_OS_NAME = osx ]; then brew install cmake gmp mpfr libmpc checkbashisms || true; fi
   
 warnings_are_errors: true
 

--- a/configure
+++ b/configure
@@ -40,12 +40,7 @@ echo == setting environment variables:
 CC=`"${R_HOME}/bin/R" CMD config CC`                   ;  echo set CC=$CC                   ; export CC
 CFLAGS=`"${R_HOME}/bin/R" CMD config CFLAGS`           ;  echo set CFLAGS=$CFLAGS           ; export CFLAGS
 CPICFLAGS=`"${R_HOME}/bin/R" CMD config CPICFLAGS`     ;  echo set CPICFLAGS=$CPICFLAGS     ; export CPICFLAGS
-
-## There is an issue (#88) compiling the C++ library when combining
-## "-std=gnu++11" and "-mmacosx-version-min=10.13". This is a temporary workaround.
 CXX=`"${R_HOME}/bin/R" CMD config CXX`                 ;  echo set CXX=$CXX                 ; export CXX
-CXX=`echo $CXX | sed 's/ -mmacosx-version-min=[^[:space:]]*//'`  ;  echo reset CXX=$CXX               ; export CXX
-
 CXXFLAGS=`"${R_HOME}/bin/R" CMD config CXXFLAGS`       ;  echo set CXXFLAGS=$CXXFLAGS       ; export CXXFLAGS
 CXXPICFLAGS=`"${R_HOME}/bin/R" CMD config CXXPICFLAGS` ;  echo set CXXPICFLAGS=$CXXPICFLAGS ; export CXXPICFLAGS
 CPPFLAGS=`"${R_HOME}/bin/R" CMD config CPPFLAGS`       ;  echo set CPPFLAGS=$CPPFLAGS       ; export CPPFLAGS
@@ -107,6 +102,7 @@ if [ -z "$SymEngine_DIR" ]; then
         -DBUILD_TESTS=no \
         -DBUILD_BENCHMARKS=no \
         -DBUILD_SHARED_LIBS=no \
+        -DWITH_COTIRE=no \
         .
 
     if [ $? -ne 0 ]; then

--- a/configure
+++ b/configure
@@ -44,7 +44,7 @@ CPICFLAGS=`"${R_HOME}/bin/R" CMD config CPICFLAGS`     ;  echo set CPICFLAGS=$CP
 ## There is an issue (#88) compiling the C++ library when combining
 ## "-std=gnu++11" and "-mmacosx-version-min=10.13". This is a temporary workaround.
 CXX=`"${R_HOME}/bin/R" CMD config CXX`                 ;  echo set CXX=$CXX                 ; export CXX
-CXX=`echo $CXX | sed 's/ -mmacosx-version-min=\S*//'`  ;  echo reset CXX=$CXX               ; export CXX
+CXX=`echo $CXX | sed 's/ -mmacosx-version-min=[^[:space:]]*//'`  ;  echo reset CXX=$CXX               ; export CXX
 
 CXXFLAGS=`"${R_HOME}/bin/R" CMD config CXXFLAGS`       ;  echo set CXXFLAGS=$CXXFLAGS       ; export CXXFLAGS
 CXXPICFLAGS=`"${R_HOME}/bin/R" CMD config CXXPICFLAGS` ;  echo set CXXPICFLAGS=$CXXPICFLAGS ; export CXXPICFLAGS

--- a/configure
+++ b/configure
@@ -44,7 +44,7 @@ CPICFLAGS=`"${R_HOME}/bin/R" CMD config CPICFLAGS`     ;  echo set CPICFLAGS=$CP
 ## There is an issue (#88) compiling the C++ library when combining
 ## "-std=gnu++11" and "-mmacosx-version-min=10.13". This is a temporary workaround.
 CXX=`"${R_HOME}/bin/R" CMD config CXX`                 ;  echo set CXX=$CXX                 ; export CXX
-CXX=`echo $CXX | sed 's/ -mmacosx-version-min=\S+//'`  ;  echo reset CXX=$CXX               ; export CXX
+CXX=`echo $CXX | sed 's/ -mmacosx-version-min=\S*//'`  ;  echo reset CXX=$CXX               ; export CXX
 
 CXXFLAGS=`"${R_HOME}/bin/R" CMD config CXXFLAGS`       ;  echo set CXXFLAGS=$CXXFLAGS       ; export CXXFLAGS
 CXXPICFLAGS=`"${R_HOME}/bin/R" CMD config CXXPICFLAGS` ;  echo set CXXPICFLAGS=$CXXPICFLAGS ; export CXXPICFLAGS

--- a/configure
+++ b/configure
@@ -40,7 +40,12 @@ echo == setting environment variables:
 CC=`"${R_HOME}/bin/R" CMD config CC`                   ;  echo set CC=$CC                   ; export CC
 CFLAGS=`"${R_HOME}/bin/R" CMD config CFLAGS`           ;  echo set CFLAGS=$CFLAGS           ; export CFLAGS
 CPICFLAGS=`"${R_HOME}/bin/R" CMD config CPICFLAGS`     ;  echo set CPICFLAGS=$CPICFLAGS     ; export CPICFLAGS
+
+## There is an issue (#88) compiling the C++ library when combining
+## "-std=gnu++11" and "-mmacosx-version-min=10.13". This is a temporary workaround.
 CXX=`"${R_HOME}/bin/R" CMD config CXX`                 ;  echo set CXX=$CXX                 ; export CXX
+CXX=`echo $CXX | sed 's/ -mmacosx-version-min=\S+//'`  ;  echo reset CXX=$CXX               ; export CXX
+
 CXXFLAGS=`"${R_HOME}/bin/R" CMD config CXXFLAGS`       ;  echo set CXXFLAGS=$CXXFLAGS       ; export CXXFLAGS
 CXXPICFLAGS=`"${R_HOME}/bin/R" CMD config CXXPICFLAGS` ;  echo set CXXPICFLAGS=$CXXPICFLAGS ; export CXXPICFLAGS
 CPPFLAGS=`"${R_HOME}/bin/R" CMD config CPPFLAGS`       ;  echo set CPPFLAGS=$CPPFLAGS       ; export CPPFLAGS
@@ -50,8 +55,8 @@ if test -z "$CXX"; then
     echo >&2 "could not detect C++ compiler with R CMD config"
 fi
 
-CMAKE_C_COMPILER=$CC    ; echo set CMAKE_C_COMPILER=$CC    ; export CMAKE_C_COMPILER
-CMAKE_CXX_COMPILER=$CXX ; echo set CMAKE_CXX_COMPILER=$CXX ; export CMAKE_CXX_COMPILER
+#CMAKE_C_COMPILER="$CC"    ; echo set CMAKE_C_COMPILER=$CC    ; export CMAKE_C_COMPILER
+#CMAKE_CXX_COMPILER="$CXX" ; echo set CMAKE_CXX_COMPILER=$CXX ; export CMAKE_CXX_COMPILER
 
 # set CMAKE_C/CXX_FLAGS and prepend CPPFLAGS
 CMAKE_C_FLAGS="$CPPFLAGS $CFLAGS"     ; echo set CMAKE_C_FLAGS="$CPPFLAGS $CFLAGS"    ; export CMAKE_C_FLAGS

